### PR TITLE
Subscriptions: Paid subscribers content importer author mapping fixes

### PR DESCRIPTION
**Pull request by pkuliga**
Related to https://github.com/Automattic/wp-calypso/issues/93150

## Proposed Changes

It makes some fixes to the upload content step:
- fixes the author mapping UI to better match the designs
- makes the "Skip for now" button disabled when uploading is in progress
- fixes the error message by removing "WordPress" word since this message is visible also when importing content from other platforms

Screenshot 2024-08-16 at 10 28 11

Screenshot 2024-08-16 at 10 26 04

## Why are these changes being made?

To make content importer UI look closer to the designs.

## Testing Instructions

*

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by bogiii — CHANGES_REQUESTED** [review]
I had a quick look at this PR, and one thing that caught my eye was that translations of the introduced string are missing.

**Comment by pkuliga** [comment]
Hi @bogiii, thanks for looking at this!

The translations are missing because the whole feature is still in a WIP state and the UI is not in a final shape. We decided not to translate strings until we agreed on their final shape.

File: client/my-sites/importer/error-pane.jsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ class ImporterError extends PureComponent {
 		}
 
 		return this.props.translate(
-			'%(errorDescription)s{{br/}}Make sure you are using a valid WordPress export file in XML or ZIP format. {{cs}}Still need help{{/cs}}?',
```
Comment (enejb) [inline client/my-sites/importer/error-pane.jsx:82]: Can we use the `this.props.importerEngine` and have more specific error messages for different engine types. 
```
if ( this.props.importerEngine === 'substack' ) {
					return this.props.translate( '%(errorDescription)s{{br/}}Make sure you are using a valid ZIP export file downloaded from substack.com. {{cs}}Still need help{{/cs}}?', ...
}
```

Comment (pkuliga) [reply client/my-sites/importer/error-pane.jsx:82]: We have it already, look at:
https://github.com/Automattic/wp-calypso/blob/d54be4cb294a8a06e76b1a8e6db0450165341091/client/my-sites/importer/error-pane.jsx#L70-L92

The problem is for imports other than from WordPress we still mention WordPress in the error message. My change just removes the "WordPress" word to make it more generic.

Comment (pkuliga) [reply client/my-sites/importer/error-pane.jsx:82]: Sure, we can adjust those error messages later but for now, I think it's good enough as it is.

Comment (enejb) [reply client/my-sites/importer/error-pane.jsx:82]: Sounds good!

**Review by enejb — APPROVED** [review]
Thanks for the changes! This tested well for me!

**Pull request merged**